src/status_im/ui/screens/discover/events.cljs

```diff
@@ -1,148 +1,172 @@
 (ns status-im.ui.screens.discover.events
-  (:require [re-frame.core :refer [after dispatch enrich]]
-            [status-im.utils.utils :refer [first-index]]
-            [status-im.utils.handlers :refer [register-handler get-hashtags]]
-            [status-im.protocol.core :as protocol]
-            [status-im.ui.screens.navigation :as nav]
+  (:require [re-frame.core :as re-frame]
             [status-im.data-store.discover :as discoveries]
-            [status-im.utils.handlers :as u]
-            [status-im.utils.datetime :as time]
-            [status-im.utils.random :as random]
-            [taoensso.timbre :as log]
+            [status-im.protocol.core :as protocol]
+            [status-im.ui.screens.discover.navigation]
             [status-im.utils.handlers :as handlers]))
 
 (def request-discoveries-interval-s 600)
 
-(register-handler :init-discoveries
-  (fn [db _]
-    (-> db
-        (assoc :tags [])
-        (assoc :discoveries {}))))
+;; EFFECTS
+
+(re-frame/reg-fx
+  ::send-portions
+  (fn [{:keys [current-public-key web3 contacts to]}]
+    (protocol/send-discoveries-response!
+     {:web3        web3
+      :discoveries (discoveries/get-all :asc)
+      :message     {:from current-public-key
+                    :to   to}})))
+
+(re-frame/reg-fx
+  ::delete-old-discoveries
+  (fn [_]
+    (discoveries/delete :created-at :asc 1000 200)))
```

Comment:
Magic numbers?

Reply:
yeah I plan to work on these data-store calls soon at least it's isolated for in an effect now